packages/scm/src/browser/scm-widget.tsx

```diff
@@ -454,8 +457,8 @@ export class ScmResourceComponent extends ScmElement<ScmResourceComponent.Props>
             onMouseEnter={this.showHover}
             onMouseLeave={this.hideHover}
             ref={this.detectHover}
-            onClick={this.selectChange}
-            onDoubleClick={this.open}>
+            onClick={(corePreferences && corePreferences['workbench.list.openMode'] === 'singleClick') ? this.open : this.selectChange}
```

Comment:
this logic should happen then someone actually clicks, otherwise if preference value is changed old event handler is still used

Reply:
I updated based on your suggestion.